plugins/chat/app/models/chat_channel.rb

```diff
@@ -21,6 +21,7 @@ class ChatChannel < ActiveRecord::Base
             },
             presence: true,
             allow_nil: true
+  validate :ensure_slug
```

Comment:
I dont like that that `validate` is validating new slug but also generating auto slug.

What would you say if we would do something like:
```
validate :slug
before_safe :generate_auto_slug

def generate_auto_slug
  return if self.slug.present?
  self.slug = Slug.for(slug_title, "")
  self.slug = "" if duplicate_slug?
end
```

Reply:
Yes that's probably a better idea and clearer, I copied from category which did the same thing, will change